src/main/java/de/tum/in/www1/artemis/service/compass/CompassCalculationEngine.java

```diff
@@ -149,37 +147,52 @@ private double getTotalConfidence() {
     }
 
     private void assessModelsAutomatically() {
-        automaticAssessmentController.assessModelsAutomatically(modelIndex, assessmentIndex);
+        automaticAssessmentController.assessModelsAutomatically(modelIndex);
     }
 
-    @Override
+    /**
+     * Get the given number of ids of the next optimal modeling submissions. Optimal means that an assessment for this model results in the biggest knowledge gain for Compass which
+     * can be used for automatic assessments.
+     *
+     * @param numberOfModels the number of next optimal models to load
+     * @return the ids of the next optimal modeling submissions, or an empty list if there are no unhandled submissions
+     */
     public List<Long> getNextOptimalModels(int numberOfModels) {
         lastUsed = LocalDateTime.now();
         return modelSelector.selectNextModels(modelIndex, numberOfModels);
     }
 
-    @Override
+    /**
+     * Get the assessment result for a model. If no assessment is saved for the given model, it tries to create a new one automatically with the existing information of the engine.
+     *
+     * @param modelSubmissionId the id of the model
+     * @return the assessment result for the model
+     */
     public Grade getGradeForModel(long modelSubmissionId) {
         lastUsed = LocalDateTime.now();
```

Comment:
You can use Java's [Clock](https://docs.oracle.com/javase/8/docs/api/java/time/Clock.html) class to simplify the testing.

You could create a private variable `clock` and instead use the following:
`LocalDateTime.now(clock)`